controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/ScaleOperationTask.java

```diff
@@ -68,9 +69,15 @@ public ScaleOperationTask(final StreamMetadataTasks streamMetadataTasks,
                         if (cause instanceof RetriesExhaustedException) {
                             cause = cause.getCause();
                         }
-                        log.warn(request.getRequestId(), "processing scale request for {}/{} segments {} failed {}",
-                                request.getScope(), request.getStream(), request.getSegmentsToSeal(), cause);
-                        result.completeExceptionally(cause);
+                        if (cause instanceof EpochTransitionOperationExceptions.PreConditionFailureException) {
+                            log.warn(request.getRequestId(), "processing scale request for {}/{} segments {} failed {}",
+                                    request.getScope(), request.getStream(), request.getSegmentsToSeal(), cause.getClass().getName());
+                            result.complete(null);
+                        } else {
+                            log.warn(request.getRequestId(), "processing scale request for {}/{} segments {} failed {}",
```

Comment:
Help me understand whether `warn` is the correct severity level here. The `cause` in this `else` block is not expected, but if we hit it, then is it something that the user must take action upon (`error`), might have to take action upon (`warn`), or only needs to be informed (`info`)?

Reply:
Processing could fail with some two classes of exceptions - retryable exception or non retryable exception. 
In case of retryable exception, the framework posts the event back into the stream so that its retried later. 
In case of non-retryable exception, the processing fails. There are some cases where a non-retryable is expected (e.g. invalid input) or a result of some code bug. in case of latter user may want to report it the error to us.

Reply:
You are also changing the behavior here besides logging. Previously you were not doing `result.complete(null)` at all.

Reply:
Yes, on receiving PreconditionFailed, we used to throw the exception which resulted in processing to be completed with the PreconditionFailed logged here and by the event processor framework too. (This is also mentioned in the issue description). 

There is nothing that we achieve by throwing that exception as the intent is to simply complete the processing for this event. So i am only just completing the processing's future by logging this at warn level without the stack trace.